akka-http-core/src/main/scala/akka/http/impl/engine/http2/client/ResponseParsing.scala

```diff
@@ -87,7 +93,11 @@ private[http2] object ResponseParsing {
         case (name, _) if name.startsWith(":") =>
           malformedRequest(s"Unexpected pseudo-header '$name' in response")
 
-        case (name, value) =>
+        case (name, httpHeader: HttpHeader) =>
+          // FIXME: move validation to HeaderDecompression validateHeader(httpHeader)
+          rec(remainingHeaders.tail, status, contentType, contentLength, seenRegularHeader = true, headers += httpHeader)
```

Comment:
This fixme is about the entire block of checking rather than this last case?

Reply:
It's about this particular case and it was indeed missing here. I moved it to `HeaderDecompression` and also changed `RequestParsing` not to re-validate there.

The advantage of doing it in `HeaderDecompression` is that we only need to validate once per header that has once been successfully parsed into a model and then stored in the HPACK tables.